### ISSUE TITLE
Loop not cleared when go()ing from inside the loop

### ISSUE THREAD

**Issue by m3g4p0p** [replit/kaboom#36]
Hey again,

I just discovered a small bug while implementing a countdown to switch scenes: when calling `go()` from within a `loop()`, that loop doesn't get cleared but keeps running. To reproduce:

```javascript
const k = kaboom()

k.scene('start', () => {
  let count = 0

  k.loop(1, () => {
    k.go('main', ++count)
  })
})

k.scene('main', count => {
  k.add([
    k.text(`Entered main scene ${count}`)
  ])
})

k.start('start')
```

Cheers! :)

**Edit:** Actually, this can also be induced by just calling `loop()` right after `go()` -- inside `wait()`, `curScene()` then already returns the next scene to which the loop getting added:

```javascript
k.scene('start', () => {
  k.go('main')
  k.loop(1, () => console.log('loop'))
})

k.scene('main', () => {
  k.add([
    k.text('Entered main scene')
  ])
})
```

**Comment by m3g4p0p** [comment]
FWIW, this might be a possible solution:

```javascript
function go(name, ...args) {
  if (game.curScene) {
    game.nextScene = (() => {
      game.curScene = null
      go(name, ...args)
    })
    return
  }
  
  reload(name);
  game.nextScene = null
  game.curScene = name;
  // ...
}
```

And at the bottom of `gameFrame()`:

```javascript
if (game.nextScene) {
  game.nextScene()
}
```

This would ensure that anything immediately following `go()` still applies to the current scene, and the actual scene switch is scheduled to the end of the frame.

**Comment by slmjkdbtl** [comment]
Nice, yeah it should happen on frame end, fixed in d3ee909, thanks!

**Comment by m3g4p0p** [comment]
Glad I could help. :)
